org.opent2t.sample.binaryswitch.superpopular/com.contosothings.binaryswitch/js/tests/test.js

```diff
@@ -0,0 +1,84 @@
+var test = require('ava');
+var OpenT2T = require('opent2t').OpenT2T;
+var config = require('./testConfig');
+
+console.log("Config:");
+console.log(JSON.stringify(config, null, 2));
+
+var translatorPath = require('path').join(__dirname, '..');
+var hubPath = require('path').join(__dirname, '../../../../org.opent2t.sample.hub.superpopular/com.contosothings.hub/js');
+var translator = undefined;
+
+function getBinarySwitch(devices) {
+    for (var i = 0; i < devices.length; i++) {
+        var d = devices[i];
+
+        if (d.opent2t.translator === 'opent2t-translator-com-contosothings-binaryswitch') {
+            return d;
+        }
+    }
+
+    return undefined;
+}
+
+// setup the translator before all the tests run
+test.before(async () => {
+    var hubTranslator = await OpenT2T.createTranslatorAsync(hubPath, 'thingTranslator', config);
+    var hubInfo = await OpenT2T.invokeMethodAsync(hubTranslator, 'org.opent2t.sample.hub.superpopular', 'get', []);
+    var deviceInfo = getBinarySwitch(hubInfo.platforms);
+
+    translator = await OpenT2T.createTranslatorAsync(translatorPath, 'thingTranslator', {'deviceInfo': deviceInfo, 'hub': hubTranslator});
+});
+
+test.serial("Valid Binary Switch Translator", t => {
+    t.is(typeof translator, 'object') && t.truthy(translator);
+});
+
+///
+/// Run a series of tests to validate the translator
+///
```

Comment:
I don't understand this comment

---
In reply to: [103512539](https://github.com/openT2T/translators/pull/175#discussion_r103512539) [](ancestors = 103512539)